File: 118.pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        vector<int> row;
        for (int i = 0; i < numRows; i++) {
            row.push_back(1);
            for (int j = row.size() - 2; j > 0; j--) {
                row[j] = row[j - 1] + row[j];
            }
            ans.push_back(row);
        }
        return ans;
    }
};

File: 118.pascals-triangle.py
class Solution:
    def generate(self, numRows: 'int') -> 'List[List[int]]':
        ans = []
        row = []
        for i in range(numRows):
            row.append(1)
            for j in range(len(row) - 2, 0, -1):
                row[j] = row[j - 1] + row[j]
            ans.append(row[:])
        return ans
